typescript/cli/src/config/ism.ts

```diff
@@ -87,116 +90,191 @@ export async function createIsmConfig(
-
-  const modules: Array<IsmConfig> = [];
-  for (let i = 0; i < isms; i++) {
-    modules.push(await createIsmConfig(context));
-  }
-  return {
-    type: IsmType.AGGREGATION,
-    modules,
-    threshold,
-  };
-}
+    const validatorsInput = await input({
+      message:
+        'Enter validator addresses (comma separated list) for message ID multisig ISM',
+    });
+    const validators = validatorsInput.split(',').map((v) => v.trim());
+    return {
+      type: IsmType.MESSAGE_ID_MULTISIG,
+      threshold,
+      validators,
+    };
+  },
+  IsmType.MESSAGE_ID_MULTISIG,
+);
 
-export async function createRoutingConfig(
-  context: CommandContext,
-  type: IsmType.ROUTING | IsmType.FALLBACK_ROUTING,
-): Promise<IsmConfig> {
-  const owner = await input({
-    message: 'Enter owner address for routing ISM',
-  });
-  const ownerAddress = owner;
-
-  const chains = await runMultiChainSelectionStep(
-    context.chainMetadata,
-    'Select chains to configure ISM for',
-    true,
-  );
-
-  const domainsMap: ChainMap<IsmConfig> = {};
-  for (const chain of chains) {
-    await confirm({
-      message: `You are about to configure ISM from source chain ${chain}. Continue?`,
+export const createTrustedRelayerConfig = callWithConfigCreationLogsAsync(
+  async (context: CommandContext): Promise<TrustedRelayerIsmConfig> => {
+    const relayer = await detectAndConfirmOrPrompt(
+      async () => context.signer?.getAddress(),
+      'For trusted relayer ISM, enter',
+      'relayer address',
+    );
+    return {
+      type: IsmType.TRUSTED_RELAYER,
+      relayer,
+    };
+  },
+  IsmType.TRUSTED_RELAYER,
+);
+
+export const createAggregationConfig = callWithConfigCreationLogsAsync(
+  async (context: CommandContext): Promise<AggregationIsmConfig> => {
+    const isms = parseInt(
+      await input({
+        message: 'Enter the number of ISMs to aggregate (number)',
+      }),
+      10,
+    );
+
+    const threshold = parseInt(
+      await input({
+        message: 'Enter the threshold of ISMs for verification (number)',
+      }),
+      10,
+    );
+
+    const modules: Array<IsmConfig> = [];
+    for (let i = 0; i < isms; i++) {
+      modules.push(await createIsmConfig(context));
+    }
+    return {
+      type: IsmType.AGGREGATION,
+      modules,
+      threshold,
+    };
+  },
+  IsmType.AGGREGATION,
+);
+
+export const createRoutingConfig = callWithConfigCreationLogsAsync(
+  async (context: CommandContext): Promise<IsmConfig> => {
+    const owner = await input({
+      message: 'Enter owner address for routing ISM',
+    });
+    const ownerAddress = owner;
+
+    const chains = await runMultiChainSelectionStep(
+      context.chainMetadata,
+      'Select chains to configure routing ISM for',
+      true,
+    );
+
+    const domainsMap: ChainMap<IsmConfig> = {};
+    for (const chain of chains) {
+      await confirm({
+        message: `You are about to configure routing ISM from source chain ${chain}. Continue?`,
+      });
```

Comment:
does not seem like we are using the result of this? should this just be a log?

Reply:
why did we separate fallback routing and routing when the prompts are the same?

Reply:
one mentions fallback, the other does not. it's just to be explicit in the CLI output about what's happening

Reply:
yeah not sure what the original intent behind this was– ended up just swapping for a log